Dockerfile

```diff
@@ -1,32 +1,45 @@
 # WebApollo
-# VERSION 2.0
-FROM ubuntu:16.10
-MAINTAINER Eric Rasche <esr@tamu.edu>, Nathan Dunn <nathandunn@lbl.gov>
+# VERSION master
+FROM tomcat:8-jre8
+MAINTAINER Anthony Bretaudeau <anthony.bretaudeau@inra.fr>, Eric Rasche <esr@tamu.edu>, Nathan Dunn <nathandunn@lbl.gov>
 ENV DEBIAN_FRONTEND noninteractive
 
 RUN apt-get -qq update --fix-missing && \
-	apt-get --no-install-recommends -y install \
-	git build-essential maven libpq-dev postgresql-common tomcat8 openjdk-8-jdk nodejs npm wget \
-	postgresql postgresql-client xmlstarlet netcat  \
-	zlib1g-dev libexpat1-dev ant curl ssl-cert && \
-	apt-get autoremove -y && apt-get clean && rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/*
+    apt-get --no-install-recommends -y install \
+    git build-essential maven2 openjdk-8-jdk libpq-dev postgresql-common \
+    postgresql-client xmlstarlet netcat libpng12-dev zlib1g-dev libexpat1-dev \
+    ant perl5 curl ssl-cert && \
+    apt-get autoremove -y && apt-get clean && rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/*
 
-RUN curl -sL https://deb.nodesource.com/setup_6.x | bash -
-RUN apt-get -qq update --fix-missing && \
-	apt-get --no-install-recommends -y install npm
+RUN curl -sL https://deb.nodesource.com/setup_6.x | bash - && \
+    apt-get -qq update --fix-missing && \
+    apt-get --no-install-recommends -y install \
+    nodejs
 
-RUN ln -s /usr/bin/nodejs /usr/bin/node && \
-	npm install -g bower && \
-	cp /usr/lib/jvm/java-8-openjdk-amd64/lib/tools.jar /usr/lib/jvm/java-8-openjdk-amd64/jre/lib/ext/tools.jar && \
-	useradd -ms /bin/bash -d /apollo apollo
+RUN npm install -g bower && \
+    cp /usr/lib/jvm/java-8-openjdk-amd64/lib/tools.jar /usr/lib/jvm/java-8-openjdk-amd64/jre/lib/ext/tools.jar && \
+    useradd -ms /bin/bash -d /apollo apollo
 
 # RUN cpan notest install Text::Markdown  # needed for apollo release
-ENV WEBAPOLLO_VERSION ac7150fdbf3ffc07d4bcf5245a5c9984b01d8ec7
+#ENV WEBAPOLLO_VERSION ac7150fdbf3ffc07d4bcf5245a5c9984b01d8ec7
+ENV WEBAPOLLO_VERSION master
```

Comment:
This change is not a good idea. There's a reason we pin version, it's so we can cache bust. The docker image will not be re-built correctly, it's undefined behaviour. There's no way to know if the builder will use a cached copy of this layer (since the layer has not changed), or if it will not have a cached layer.

The only time it's really appropriate, is if the dockerfile was part of the gmod/apollo github repo. Then you would normally setup a docker build step that would build on push of tags / push to master. That's a very different case though, and you wouldn't even use `git clone`, there, you'd use `ADD . /apollo/` (or similar).